test/getConfig.spec.js

```diff
@@ -0,0 +1,242 @@
+        renderer: 'update'
+      })
+    )
+
+    expect(
+      getConfig({
+        verbose: true
+      })
+    ).toEqual(
+      expect.objectContaining({
+        renderer: 'verbose'
+      })
+    )
+  })
+
+  it('should set gitDir', () => {
+    expect(getConfig({})).toEqual(
+      expect.objectContaining({
+        gitDir: '.'
+      })
+    )
+
+    expect(
+      getConfig({
+        gitDir: '../path'
+      })
+    ).toEqual(
+      expect.objectContaining({
+        gitDir: '../path'
+      })
+    )
+  })
+
+  it('should set linters', () => {
+    expect(getConfig()).toEqual(
+      expect.objectContaining({
+        linters: {}
+      })
+    )
+
+    expect(getConfig({})).toEqual(
+      expect.objectContaining({
+        linters: {}
+      })
+    )
+
+    expect(
+      getConfig({
+        '*.js': 'eslint'
+      })
+    ).toEqual(
+      expect.objectContaining({
+        linters: {
+          '*.js': 'eslint'
+        }
+      })
+    )
+
+    expect(
+      getConfig({
+        linters: {
+          '*.js': ['eslint --fix', 'git add'],
+          '.*rc': 'jsonlint'
+        }
+      })
+    ).toMatchSnapshot()
+  })
+
+  it('should deeply merge configs', () => {
+    expect(
+      getConfig({
+        globOptions: {
+          nocase: true
+        }
+      })
+    ).toEqual(
+      expect.objectContaining({
+        globOptions: {
+          nocase: true,
+          matchBase: true,
+          dot: true
+        }
+      })
+    )
+  })
+
+  it('should not add plain linters object to the full config', () => {
+    expect(
+      getConfig({
+        '*.js': 'eslint'
+      })
+    ).not.toEqual(
+      expect.objectContaining({
+        '*.js': 'eslint'
+      })
+    )
+  })
+
+  it('should not change config if the whole config was passed', () => {
+    const src = {
+      concurrent: false,
+      chunkSize: 2,
+      gitDir: '/to',
+      globOptions: {
+        matchBase: false,
+        dot: true
+      },
+      linters: {
+        '*.js': 'eslint'
+      },
+      subTaskConcurrency: 10,
+      renderer: 'custom',
+      verbose: true
+    }
+    expect(getConfig(src)).toEqual(src)
+  })
+})
+
+describe('validateConfig', () => {
+  beforeEach(() => {
+    global.console = makeConsoleMock()
+  })
+
+  it('should throw and print validation errors for invalid config', () => {
+    const invalidAdvancedConfig = {
+      foo: false,
+      chunkSize: 'string',
+      gitDir: 111
+    }
+    expect(() => validateConfig(getConfig(invalidAdvancedConfig))).toThrowErrorMatchingSnapshot()
+  })
+
+  it('should not throw and print validation warnings for mixed config', () => {
+    const invalidMixedConfig = {
+      gitDir: './path/to/packagejson/',
+      '*.js': ['eslint --fix', 'git add']
+    }
+    expect(() => validateConfig(getConfig(invalidMixedConfig))).not.toThrow()
+    expect(console.printHistory()).toMatchSnapshot()
+  })
+
+  it('should not throw and print nothing for valid config', () => {
```

Comment:
A small gripe but this test title is ambiguous. Should it not throw and _not_ print nothing? `should not throw and should print nothing for valid config` is clearer.